File: solutions/clojure/armstrong-numbers/2/src/armstrong_numbers.clj
(ns armstrong-numbers)

(defn armstrong? [num]
  (let [digits (->> (take-while 
                     (partial < 0) 
                     (iterate #(quot % 10) num))
                    (map #(rem % 10)))
        pow (fn [base exp] (apply * (repeat exp base)))
        num-digits (count digits)
        sum-digits (->> digits
                        (map #(pow % num-digits))
                        (apply +))]
    (= num sum-digits)))

File: solutions/cpp/armstrong-numbers/2/armstrong_numbers.cpp
#include "armstrong_numbers.h"
#include <vector>
#include <cmath>
#include <iostream>

namespace armstrong_numbers {
    bool is_armstrong_number(int n) {
        int nCopy{n};
        int sum{};
        int count = std::log10(n) + 1;
        
        while (n > 0) {
            div_t div_r = std::div(n, 10);
            sum += std::pow(div_r.rem, count);
            n = div_r.quot;
        }

        return sum == nCopy;
    }
}  // namespace armstrong_numbers
